source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -252,23 +238,50 @@ namespace
 
 const string OgsXmlGenerator::OUTPUT_NAME = "outColor";
 const string OgsXmlGenerator::VP_TRANSPARENCY_NAME = "vp2Transparency";
-const string OgsXmlGenerator::SAMPLER_SUFFIX = "Sampler";
+const string OgsXmlGenerator::SAMPLER_SUFFIX = "_sampler";
 
-OgsXmlGenerator::OgsXmlGenerator()
+bool OgsXmlGenerator::isSamplerName(const string& name)
 {
+    // We follow the SPIRV-Cross naming conventions for HLSL samplers
+    // auto-generated from combined GLSL sampers. This happens to be compatible
+    // with the OGS convention for pairing samplers and textures which only
+    // requires that the texture name be a substring of the sampler name.
+    
+    static const size_t SUFFIX_LENGTH = SAMPLER_SUFFIX.size();
+    return name.size() > SUFFIX_LENGTH + 1
+        && name[0] == '_'
+        && 0 == name.compare(name.size() - SUFFIX_LENGTH, SUFFIX_LENGTH, SAMPLER_SUFFIX);
 }
 
-void OgsXmlGenerator::generate( const std::string& shaderName, const Shader* glsl, const Shader* hlsl,
-                                bool hwTransparency, std::ostream& stream)
+string OgsXmlGenerator::textureToSamplerName(const string& textureName)
 {
-    if (glsl == nullptr && hlsl == nullptr)
-    {
-        throw ExceptionShaderGenError("Both GLSL and HLSL shaders are null, at least one language must be given to generate XML fragments");
-    }
+    string result = "_";
+    result += textureName;
+    result += SAMPLER_SUFFIX;
+    return result;
+}
 
+string OgsXmlGenerator::samplerToTextureName(const string& samplerName)
+{
+    static const size_t PREFIX_SUFFIX_LENGTH = SAMPLER_SUFFIX.size() + 1;
+    return isSamplerName(samplerName)
+        ? samplerName.substr(1, samplerName.size() - PREFIX_SUFFIX_LENGTH) : "";
+}
```

Comment:
Changing the sampler naming convention to match `SPIRV-Cross` while maintaining compatibility with OGS.